fracking/js/lab-integration.coffee

```diff
@@ -0,0 +1,88 @@
+window.setupLabCommunication = (model) ->
+  phone = iframePhone.getIFrameEndpoint()
+
+  # Register Scripting API functions.
+  registerModelFunc = (name) ->
+    phone.addListener name, ->
+      model[name].apply(model, arguments)
+      model.draw()
+
+    phone.post('registerScriptingAPIFunc', name)
+
+  registerCustomFunc = (name, func) ->
+    phone.addListener(name, func)
+    phone.post('registerScriptingAPIFunc', name)
+
+  registerCustomFunc 'play', ->
+    model.start()
+
+  registerCustomFunc 'stop', ->
+    model.stop()
+
+  registerCustomFunc 'step', (content) ->
+    steps = content
+    while (steps--)
+      model.step()
+    model.draw()
+
+  registerModelFunc('explode')
+  registerModelFunc('floodWater')
+  registerModelFunc('floodPropane')
+  registerModelFunc('pumpOut')
+
+  # Properties.
+  phone.addListener 'set', (content) ->
+    switch content.name
+      when 'leaks'
+        model.leaks = content.value
+      when 'drillDirection'
+        model.drillDirection = content.value
+
+
+  getOutputs = ->
+    result =
+      year: model.getYear(),
+      removeFluidPossible: false,
+      fillPossible: false,
```

Comment:
The commas are unnecessary here, but that's a style choice (I tend to leave them out in these situations)...